### PR TITLE
[Stacked, 2/2] Adds a Mesh tutorial on I/O, Serialization, and Interop

### PR DESCRIPTION
**Pull request by peterdsharpe**
# PhysicsNeMo Pull Request

## Description

**NOTE: this is a stacked PR on top of https://github.com/NVIDIA/physicsnemo/pull/1711**.

Only diff from this PR is in `examples/minimal/mesh` - specifically, the addition of a new Jupyter notebook that documents:
a) PyVista interop
b) Serialization/deserialization (introducing the `*.pmsh` and `*.pdmsh` formats for `Mesh`es and `DomainMesh`es, respectively)
c) Some details about tessellation when converting to simplicial meshes.

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: docs/api/mesh/tessellation.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    # Low-level: triangle connectivity plus the parent-polygon index.
+    result = triangulate_polygons(points, connectivity, offsets)
+    result.cells          # tensor([[0, 1, 2], [0, 2, 3]])
+    result.parent_index   # tensor([0, 0]); broadcast data via cell_data[parent_index]
+
+    # High-level: build a Mesh directly, broadcasting per-polygon cell data.
+    mesh = Mesh.from_polygons(
+        points, connectivity, offsets, cell_data={"pressure": torch.tensor([2.5])}
+    )
```
[the remark this answers is not in the document]
Comment (peterdsharpe) [reply docs/api/mesh/tessellation.rst]: Whoops! Updated the API and forgot to update the docs. Will fix.

**Review by ktangsali — APPROVED** [review]
LGTM. Most comments on the first PR

File: docs/api/mesh/tessellation.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (peterdsharpe) [reply docs/api/mesh/tessellation.rst]: Fixed in #1711

**Comment by peterdsharpe** [comment]
/ok to test https://github.com/NVIDIA/physicsnemo/pull/1712/commits/ceb86c89a2e3f929172e88d27b9f9fda9eb60f7a
